### PR TITLE
CNS-815: limit projects in plan

### PR DESCRIPTION
**Pull request by oren-lava**
In this PR I introduce a new property of a plan: `projects`. This determines the max number of projects a subscription is allowed to hold.

Note, a migrator was implemented to change current plans' projects field according to the product team's docs.

### CODE REVIEW

File: proto/lavanet/lava/plans/plan.proto [hunk 1]
```diff
@@ -23,6 +23,7 @@ message Plan {
     string type = 12 [(gogoproto.jsontag) = "type"]; // plan type
     uint64 annual_discount_percentage = 13 [(gogoproto.jsontag) = "annual_discount_percentage"]; // discount for buying the plan for a year
     Policy plan_policy = 14 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "plan_policy"];
+    uint64 projects = 15 [(gogoproto.jsontag) = "projects"]; // number of allowed projects
```
Comment (Yaroms) [inline proto/lavanet/lava/plans/plan.proto:26]: can you rename this?
maybe projects_limit?

File: x/plans/keeper/migrations.go [hunk 2]
```diff
@@ -154,3 +155,74 @@ func (m Migrator) Migrate7to8(ctx sdk.Context) error {
 
 	return nil
 }
+
+func (m Migrator) Migrate8to9(ctx sdk.Context) error {
+	adventurerLimit := uint64(1)
```
Comment (Yaroms) [inline x/plans/keeper/migrations.go:160]: how did you decide those numbers?

File: x/plans/types/plan.go [hunk 3]
```diff
@@ -45,6 +45,10 @@ func (p Plan) ValidatePlan() error {
 		return sdkerrors.Wrap(ErrInvalidPlanAnnualDiscount, "plan's annual discount is invalid (not between 0-100 percent)")
 	}
 
+	if p.GetProjects() == 0 {
```
Comment (Yaroms) [inline x/plans/types/plan.go:48]: lets do 0 is unlimited

File: x/plans/keeper/migrations.go (hunk 2 above)
Comment (oren-lava) [reply x/plans/keeper/migrations.go:160]: using Sid's spreadsheet (from the subscription rates design)

File: proto/lavanet/lava/plans/plan.proto (hunk 1 above)
Comment (oren-lava) [reply proto/lavanet/lava/plans/plan.proto]: done. see https://github.com/lavanet/lava/commit/d8cbc30eb943844a5306b0300c9dbd7bcc45bb45

File: x/plans/keeper/migrations.go (hunk 2 above)
Comment (Yaroms) [reply x/plans/keeper/migrations.go:160]: the gateway needs unlimited projects as of now.
this is the whale plan, the rest you can leave as it is

File: x/plans/types/plan.go (hunk 3 above)
Comment (oren-lava) [reply x/plans/types/plan.go]: done. see https://github.com/lavanet/lava/commit/dd2aede1daae58f2e1605969f08f7679938d3b8e

**Review by Yaroms — APPROVED** [review]

**Pull request merged**